docs/architecture/transaction.md

```diff
@@ -0,0 +1,64 @@
+# Transaction
+
+> Single account state transition
+
+## What is the purpose of a transaction?
+
+To facilitate single account state transitions.
+
+## What is a transaction?
+
+In Miden, a `Transaction` represents the state transition of a single account. A `Transaction` takes a single [account](accounts.md) and one or more [notes](notes.md), and none or one script (piece of code executed after all notes have been executed) as input, and outputs the same account with a potentially updated state, together with some potential newly created notes.
```

Comment:
Slight inaccuracy: a transaction can take zero input notes (i.e., not "one or more").